src/api/app/models/bs_request.rb

```diff
@@ -89,6 +89,7 @@ class SaveError < APIError
   has_many :target_project_objects, through: :bs_request_actions
 
   belongs_to :staging_project, class_name: 'Project', foreign_key: 'staging_project_id'
+  has_one :request_ignoration, foreign_key: 'bs_request_id', dependent: :destroy
```

Comment:
`s/:request_ignoration/:ignored_request/` please...

Reply:
The **"thing"** in the middle  that we use to mark the request as ignored is called `RequestIgnoration`. So, the "problem" is the name of the class and not the name of the association.

Please, if any of you have a better name please say it :pray:

Reply:
A request has_one :ignore_reason ? :)

Reply:
I think the problem naming this thing comes from the fact that you think that ignoring requests is something generic. It's not, it's something specific for the staging workflow. This **thing** is similar to `StagedRequest`. For instance:

- The request is staged: `StagingProject` <-> `StagedRequest` <-> `BsRequest`
- The request is ignored for staging: `StagingWorkflow` <-> `StagingIgnored` <-> `BsRequest`

Reply:
Fine by my too.

Reply:
> I like `IgnoreReason` the only thing I see is that the model represents the association, not only the reason.
> 
> My proposal is that we can use [`exclusion`](https://en.wiktionary.org/wiki/exclusion) instead of **ignoration.** In the end we are excluding request to not take them into account.
> 
> Then the class would be `Staging::RequestExclusion`, the direct association would be `request_exclusions` and the through association would be `excluded_requests` (that will return the `BsRequest` objects).

Done.